File: Data Structures/Array/1 D/Memorise Me.cpp
//https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/memorise-me/
#include<iostream>
#include<cstdlib>
using namespace std;
long a[2][100000];

int main()
    {
        long n,i,q;
        int temp;
        cin>>n;
        
        for(i=0;i<n;i++)
            {
                cin>>temp;
                if(temp>=0)
                    a[0][temp]++;
                else
                    a[1][abs(temp)]++;
            }
        cin>>q;
        
        for(i=0;i<q;i++)
            {
                cin>>temp;
                if(temp>=0 && a[0][temp]!=0)
                    cout<<a[0][temp]<<endl;
                else if(temp<0 && a[1][temp]!=0)
                    cout<<a[1][temp]<<endl;
                else
                    cout<<"NOT PRESENT"<<endl;
            }
        
        return 0;
    }

File: Data Structures/Array/1 D/Memorise Me.py
#https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/memorise-me/
n=int(input())
a = input().split(" ")
x=[]
y=[]
for i in range(100000):
    x.append(0)
    y.append(0)
for i in range(n):
    if int(a[i]) >= 0:
        x[int(a[i])]+=1
    else :
        y[abs(int(a[i]))]+=1

q = int(input())
temp = 0
for i in range(q):
    temp = int(input())
    if (temp >= 0 and int(x[temp]) != 0 ):
        print(x[temp])
    elif (temp < 0 and int(x[temp]) != 0 ):
        print(y[temp])
    else :
        print("NOT PRESENT")